src/components/ScheduleFormModal.vue

```diff
@@ -47,19 +52,37 @@
 
   const props = defineProps<DeploymentScheduleCompatible>()
 
+  const can = useCan()
+
   const internalActive = ref<boolean>(props.active ?? true)
 
+  const { validate } = useValidationObserver()
+  const internalJobVariables = ref<string | undefined>(props.jobVariables ? stringify(props.jobVariables) : undefined)
+
   const emit = defineEmits<{
     (event: 'submit', value: DeploymentScheduleCompatible): void,
   }>()
 
 
-  const scheduleFormSubmit = (schedule: Schedule): void => {
-    submit({ 'active': internalActive.value, 'schedule': schedule })
-  }
+  async function submit(schedule: Schedule | null): Promise<void> {
+    const valid = await validate()
+
+    if (!valid) {
+      return
+    }
 
-  const submit = (schedule: DeploymentScheduleCompatible): void => {
-    emit('submit', schedule)
+    let jobVariables: Record<string, unknown> | undefined
+    if (!can.access.flowRunInfraOverrides) {
+      jobVariables = undefined
+    } else {
+      jobVariables = internalJobVariables.value ? JSON.parse(internalJobVariables.value) : undefined
+    }
```

Comment:
is stuff like this only necessary during development and will eventually get cleaned up or is this something we'll have to deal with for backwards compatibility for a while?

Reply:
the former - this is the "don't send the extra parameter to the api". in hindsight mb i could've done this in the mapper and only have 1 place to update.. 

hoping this doesn't have to last too long.